SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LegacyLoadBalancingStrategy.cpp

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/LegacyLoadBalancingStrategy.h"
+
+#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "Interop/Connection/SpatialOSWorkerInterface.h"
+#include "LoadBalancing/AbstractLBStrategy.h"
+#include "LoadBalancing/LBDataStorage.h"
+#include "LoadBalancing/LoadBalancingCalculator.h"
+#include "LoadBalancing/PartitionManager.h"
+
+namespace SpatialGDK
+{
+FLegacyLoadBalancing::FLegacyLoadBalancing(UAbstractLBStrategy& LegacyLBStrat, SpatialVirtualWorkerTranslator& InTranslator)
```

Comment:
One things that keeps tripping me up, is the responsibilities of each of the load balancing classes. It would be good to get a sentence or two on each.
For instance I don't follow why FLoadBalancingStrategy and FLoadBalancingCalculator are separate classes as their functionality is very much aligned, and it doesn't seem like you could mix and match different instances very easily. Also what their inclusion means for UAbstractLBStrategy and it's descendants,

Reply:
FLoadBalancingCalculator is just something I made to help me support the legacy strategies, but after thinking about it, I want to get rid of it. So the entire algorithm will be implemented in FLegacyLoadBalancing. How a user would want to organize their own strategy will be up to them.